src/meta/src/barrier/progress.rs

```diff
@@ -318,13 +385,35 @@ impl CreateMviewProgressTracker {
                 progress.update(actor, new_state, upstream_total_key_count);
 
                 if progress.is_done() {
-                    tracing::debug!("all actors done for creating mview with epoch {}!", epoch);
+                    tracing::debug!(
+                        "all actors done for creating mview with table_id {}!",
+                        table_id
+                    );
 
                     // Clean-up the mapping from actors to DDL epoch.
                     for actor in o.get().0.actors() {
                         self.actor_map.remove(&actor);
                     }
-                    Some(o.remove().1)
+                    match o.remove().1 {
+                        StreamJobOrigin::TrackingCommand(t) => Some(t),
+                        StreamJobOrigin::RecoveredStreamJob(Notifier {
+                            finished: Some(sender),
+                            ..
+                        }) => {
+                            if let Err(e) = sender.send(()) {
```

Comment:
Directly notify success to mark the job as completed might have some in-consistency issue, the states of recovered job could not be flushed if a recovery immediately follows. Compared to the normal tracking processing, if the progress is done the command will be stashed as finish commands in checkpoint controller and is not considered as complete until a ckpt barrier is encountered.

Reply:
And it seems like `mark_table_fragments_created` is never called for recovered jobs.

Reply:
> 

Directly notify success to mark the job as completed might have some in-consistency issue, the states of recovered job could not be flushed if a recovery immediately follows. Compared to the normal tracking processing, if the progress is done the command will be stashed as finish commands in checkpoint controller and is not considered as complete until a ckpt barrier is encountered.

I see, got your idea. So probably need to return a tracking command to the checkpoint controller.
It will also take care of `mark_table_fragments_created`.

Reply:
+1. We should avoid splitting the shared code logic into multiple places as much as possible. We may return an `enum` that is "finishable".